unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestController.cs

```diff
@@ -154,13 +164,25 @@ public static WebRequestController Create()
             int requestAttemps,
             int timeout,
             CancellationToken cancellationToken,
-            Dictionary<string, string> headers = null) where T : IWebRequestFactory
+            Dictionary<string, string> headers = null,
+            bool isSigned = false) where T : IWebRequestFactory
         {
             requestAttemps = Mathf.Max(1, requestAttemps);
+
             for (int i = requestAttemps - 1; i >= 0; i--)
             {
                 UnityWebRequest request = requestFactory.CreateWebRequest(url);
                 request.timeout = timeout;
+                Enum.TryParse(request.method, out RequestMethod method);
+
+                if (isSigned && rpcSignRequest != null)
+                {
+                    headers ??= new Dictionary<string, string>();
```

Comment:
I would omit it as it will create an instance of `Dictionary` almost in every case as we don't really pass custom headers: I would call `request.SetRequestHeader` directly